src/sentry/api/endpoints/organization_member_invite/utils.py

```diff
@@ -9,3 +10,12 @@ class MemberInviteDetailsPermission(OrganizationPermission):
         # you added, or any member invite / invite request if you have the required scopes
         "DELETE": ["member:read", "member:write", "member:admin"],
     }
+
+    def has_object_permission(self, request, view, organization):
+        """
+        Prevents superuser read from deleting an invite or invite request.
+        """
+        has_perms = super().has_object_permission(request, view, organization)
+        if is_active_superuser(request) and not superuser_has_permission(request):
```

Comment:
```suggestion
        if not superuser_has_permission(request):
```
`superuser_has_permission` already has `is_active_superuser` check inside

Reply:
I think it needs to be defined the way it is now. If we do 
```
if superuser_has_permission:
    return True
return has_perms
```
within object_has_permission, then we return True for has_perms (because superuser has read permission) but don't do any checks within the delete method to prevent superuser read from deleting.